recipe/build.sh

```diff
@@ -1,46 +1,153 @@
-#!/bin/bash
-export LD_LIBRARY_PATH="$PREFIX/lib:$LD_LIBRARY_PATH"
-export LIBRARY_PATH="$PREFIX/lib:$LIBRARY_PATH"
-export C_INCLUDE_PATH="$PREFIX/include:$C_INCLUDE_PATH"
-export LDFLAGS=" -lgmp $LDFLAGS"
-echo $LDFLAGS
-export LD="x86_64-conda_cos6-linux-gnu-ld"
-echo "Content PREFIX bin"
-ls -lrt $PREFIX/bin
-echo "Content PREFIX lib"
-ls -lrt $PREFIX/lib
-echo "Content PREFIX bin"
-ls -lrt $BUILD_PREFIX/bin
-ghc-pkg recache
-ghc-pkg describe rts
-ghc-pkg describe rts > rts.pkg
-perl -pi -e 's/$PREFIX\/lib\/ghc-8.2.2\/rts/$PREFIX\/lib\/ghc-8.2.2\/rts \$\{pkgroot\}\/../g' rts.pkg
-cat rts.pkg
-ghc-pkg update rts.pkg
-echo "Setting x86_64-conda_cos6-linux-gnu-gcc"
-echo "GCC version"
-x86_64-conda_cos6-linux-gnu-gcc --version
-echo "CC version"
-x86_64-conda_cos6-linux-gnu-cc --version
-echo "Collect2"
-x86_64-conda_cos6-linux-gnu-cc -print-prog-name=collect2
-echo "ld"
-x86_64-conda_cos6-linux-gnu-cc -print-prog-name=ld
-echo "LD version"
-x86_64-conda_cos6-linux-gnu-ld --version
-echo "LD help"
-x86_64-conda_cos6-linux-gnu-ld --help
-export EXTRA_CONFIGURE_OPTS=" --extra-include-dirs=$PREFIX/include --extra-lib-dirs=$PREFIX/lib ";
-sed -i -- 's/collect2 //g' cabal-install/bootstrap.sh
-
-ghc-pkg recache
-cd cabal-install
-echo "Extra configure opts"
-echo "$EXTRA_CONFIGURE_OPTS"
-sed -i -- 's/export LD=$LINK/export LINK=x86_64-conda_cos6-linux-gnu-cc/g' bootstrap.sh
-
-sed -i -- 's/${GHC} --make ${JOBS} ${PKG_DBS} Setup -o Setup/${GHC} -lgmp -threaded -pgmc x86_64-conda_cos6-linux-gnu-cc -pgml x86_64-conda_cos6-linux-gnu-cc --make ${JOBS} ${PKG_DBS} Setup -o Setup/g' bootstrap.sh
-cat bootstrap.sh
-export GHC=`which ghc`
-strings $GHC
-./bootstrap.sh --no-doc
+#!/usr/bin/env bash
+
+set -eux
+
+# Environment setup
+export PKG_CONFIG_PATH="${PREFIX}/lib/pkgconfig:${PKG_CONFIG_PATH:+:}${PKG_CONFIG_PATH:-}"
+export LD_LIBRARY_PATH="${PREFIX}/lib:${BUILD_PREFIX}/lib${LD_LIBRARY_PATH:+:}${LD_LIBRARY_PATH:-}"
+export LIBRARY_PATH="${PREFIX}/lib:${BUILD_PREFIX}/lib${LIBRARY_PATH:+:}${LIBRARY_PATH:-}"
+export PATH="${BUILD_PREFIX}/ghc-bootstrap/bin${PATH:+:}${PATH:-}"
+
+unset build_alias
+unset host_alias
+
+# Determine fallback URL based on platform
+get_fallback_url() {
```

Comment:
Please add this to the `source:` section of the recipe instead of manually downloading it here.